infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -127,10 +121,9 @@ public void shutdown() {
         executor.shutdown();
     }
 
-    private boolean isBootstrapped(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
+    private boolean isBootstrapped(RequestMsg requestMsg) {
```

Comment:
Notice that this check is called on all the handlers except for the actual bootstrap call. I would recommend that this check is executed in the entry point, i.e, processRequest.

```java
@Override
    protected void processRequest(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
        executor.submit(() -> {


            if (!req.getPayload().getPayloadCase().equals(RequestPayloadMsg.PayloadCase.BOOTSTRAP_LAYOUT_REQUEST)
            && !isBootstrapped(req)) {
                r.sendNoBootstrapError(req.getHeader(), ctx);
                return;
            }

            // Don't process layout server requests unless the server is bootstrapped 
            getHandlerMethods().handle(req, ctx, r);
        });
    }
```

Reply:
Yeah it's called for most handlers. But @zfrenette formerly brought up a concern that, by this way we are performing the check before the task is submitted to the executor service, and the status could change by the time the new thread is actually executing the handler. (https://github.com/CorfuDB/CorfuDB/pull/2823#discussion_r523764871) I think it's reasonable, but would like to know your opinion for this.

Reply:
but hmm followed by your next comment, those handlers are literally single threaded, then it should be fine to move this check to `processRequest`

Reply:
The status cannot change if only one thread is executing all handlers